source/MapPanel.cpp

```diff
@@ -1403,11 +1415,38 @@ void MapPanel::DrawSystems()
 		closeGovernments.clear();
 
 	// Draw the circles for the systems.
+	BatchDrawList starBatch;
 	double zoom = Zoom();
 	for(const Node &node : nodes)
 	{
 		Point pos = zoom * (node.position + center);
-		RingShader::Draw(pos, OUTER, INNER, node.color);
+		if(commodity != SHOW_STARS)
+			RingShader::Draw(pos, OUTER, INNER, node.color);
+		else
+		{
+			// Ensures every multiple-star system has a characteristic, deterministic rotation.
+			Angle starAngle = 0;
+			Angle angularSpacing = 0;
+			Point starOffset = Point(0, 0);
+
+			const int starsToDraw = min(static_cast<int>(node.mapIcons.size()), MAX_STARS);
```

Comment:
This behaves the same and matches the pattern used elsewhere in the codebase.
```suggestion
			const int starsToDraw = min<int>(node.mapIcons.size(), MAX_STARS);
```
(Yes, I do see that there are other locations in this file that use `static_cast<int>` instead of `min<int>`. You should change those, too. You can blame the 2020 version of my for merging #5555 without seeing this.)